patterns/1-initial/innersource-hackathon.md

```diff
@@ -0,0 +1,80 @@
+
+The company wants to adopt InnerSource as software development methodology but only those familiar with open source principles or those who understand the benefits of InnerSource, adopt it. It results in just a handful of InnerSource projects and is difficult to scale beyond that.
+
+## Context
+
+Scenario 1:
+The senior leadership believes in InnerSource and wants to drive it throughout the company. The engineers who are familiar with open source principles and/or understand the benefits of InnerSource are the early adopters. There is success with these initial pilot project and teams. Now the next step is to drive it across the company. There might be reluctance from engineering teams due to various factors like:
+
+* not familiar with InnerSource and being ignorant to know about it
+* not enough time to prioritize InnerSource, given the regular work deliverables
+* relunctance to changing ways of working when everything works well already
+* perception that InnerSource requires more work and responsibilities
+
+Scenario 2:
+Teams slowly start adopting InnerSource and open up their repositories. Some new projects are started as InnerSource projects from the start (that is open for contributions). However, there are not many contributors. It is also challenging to build a community around those projects, in spite of publishing the projects in an InnerSource Portal.  This could be due to various reasons like:
+
+* engineers do not have time to explore new InnerSource projects and contribute, outside regular work deliverables
+* no additional incentive to this effort apart from being acknowledged
+* lack of motivation by middle management
+
+## Forces
+
+Apart from the reasons listed in the Context section, there are other factors from different entities that prevent teams from adopting InnerSource like:
+
+* From middle management: perception of no direct benefit to the team apart from team upskilling and individual growth; eventually resulting in no motivation by the middle management to the engineers in trying out InnerSource ways of working.
+* From engineers: investing time to try new ways of working when there is already time and resource constraint for regular work deliverables; it is also challenging to balance both and show outcomes in both; this curbs motivation from the engineers as there is little safe space to experiment.
+* From senior leadership: not prioritizing investment in InnerSource ways of working such as it being part of long term goals or OKRs.
+
+For those new to InnerSource or any technology/methodology, there is a need for safe space to try and experiment without definite outcomes in the initial stages. These factors and forces hinder such an experiment ground.
+
+## Solution
+
+Organize a company wide hackathon focused on InnerSource. An InnerSource hackathon provides a safe space for engineers to try and contribute to InnerSource projects without any presumption and prejudice. It could be a 1 or 2 day event.
```

Comment:
Thinking of different organisational sizes - do you think there is a sweet spot where company wide makes sense vs. at which point it gets too large to be effective?

Reply:
I agree to your point - maybe it can be conducted as a business unit level in large organizations?

Reply:
Also depends on what you want to achieve with InnerSource in the first place.

e.g. if the purpose is to encourage re-use across department borders in a large multi-national org, then you would likely want the hackathon to span all those departments. Even if that means that the hackathon becomes larger.

Different hackathon formats and sizes have different pros and cons. But I think all of them can be effective. Sometimes you have to "encourage then good" e.g. for the prior example of the multi-national org, if you want cross-team collaboration to happen, then you can setup soft rules like "only teams that contain members from at least 2 different teams from 2 separate engineering offices are considered when choosing an overall winner".